Tools/TestWebKitAPI/Tests/WebKitCocoa/Badging.mm

```diff
@@ -462,4 +485,46 @@ - (void)websiteDataStore:(WKWebsiteDataStore *)dataStore workerOrigin:(WKSecurit
     EXPECT_EQ(badgeDelegate.get().appBadgeIndex, 2);
     EXPECT_EQ(badgeDelegate.get().clientBadgeIndex, 0);
 }
+
+static constexpr auto originMainFrameBytes = R"SWRESOURCE(
+<iframe src="origintest2://main.com/"></iframe>
+)SWRESOURCE"_s;
+
+static constexpr auto originIFrameBytes = R"SWRESOURCE(
+<script>
+window.navigator.setAppBadge(10);
+</script>
+)SWRESOURCE"_s;
+
+TEST(Badging, Origin)
+{
+    auto configuration = adoptNS([[WKWebViewConfiguration alloc] init]);
+
+    auto handler = [[TestURLSchemeHandler new] autorelease];
```

Comment:
My understanding is that we're trying to avoid explicit autorelease like this and use RetainPtr instead, to make the transition to ARC easier. so I would `adoptNS()` here, just like you did above for the configuration.